Sources/Async/BackendService+Async.swift

```diff
@@ -0,0 +1,45 @@
+//
+//  BackendService+Async.swift
+//  Hyperspace-iOS
+//
+//  Created by Daniel Larsen on 10/30/21.
+//  Copyright © 2021 Bottle Rocket Studios. All rights reserved.
+//
+
+import Foundation
+
+@available(iOS 13.0, *)
+extension BackendServiceProtocol {
+
+    /// Executes the Request asynchronously, returning the specified value type. Throws an error if unable to return the value as specified.
+    ///
+    /// - Parameters:
+    ///   - request: The Request to be executed.
+    /// - Returns: The decodable type specified in the `Request`.
+    public func execute<T, U>(request: Request<T, U>) async throws -> T {
+        return try await withCheckedThrowingContinuation { continuation in
+            execute(request: request) { result in
+                switch result {
+                case .success(let value): continuation.resume(returning: value)
+                case .failure(let err): continuation.resume(throwing: err)
+                }
+            }
+        }
+    }
+
+    /// Executes the Request asynchronously without throwing any errors
+    ///
+    /// - Parameters:
+    ///   - request: The Request to be executed.
+    /// - Returns: A `Result` of given value type `T` and error type `U`.
+    public func executeWithResult<T, U>(request: Request<T, U>) async -> Result<T, U> {
```

Comment:
I ended up giving preference to `public func execute<T, U>(request: Request<T, U>) async throws -> T` over this function when naming. Since both functions have the same signatures, renaming one is required for disambiguation. 

To be honest, I was surprised that it wasn't enough to disambiguate with `throw`ing vs non-throwing. But the compiler was complaining. 

I'm open to any suggestions for better naming!

Reply:
Ah yea, makes sense - the compiler can't tell you whether you want a `T` or `Result<T, U>` unless you were to explicitly type the output. I'm ok with the difference in names, to be honest (at the very least, I can't think of anything better!)